kf_lib_data_ingest/etl/extract/extract.py

```diff
@@ -388,86 +382,27 @@ def _run(self):
-        # containing them
-        # counted = {
-        #     a_key: {  # e.g. PARTICIPANT.ID
-        #         a1: [file1, file2],  # e.g. PARTICIPANT.ID a1 is in files 1&2
-        #         ...
-        #     },
-        #     ...
-        # }
-        counted = defaultdict(
+        sources = defaultdict(
+            lambda: defaultdict(set)
+        )
+        links = defaultdict(
             lambda: defaultdict(set)
         )
 
         for config_path, (data_file, df) in run_output.items():
             for key in df.columns:
                 for val in df[key]:
-                    counted[key][val].add(data_file)
-
-        uniques = {
-            key: len(unique_vals) for key, unique_vals in counted.items()
-        }
-
-        # display unique counts
-
-        message = ['UNIQUE COUNTS']
-        message.append(
-            pformat(uniques)
-        )
-        message.append('')
-
-        # check expected counts
-
-        all_checks_passed = True
-        message.append('EXPECTED COUNT CHECKS')
-
-        if not self.expected_counts:
-            message.append("No expected counts registered. ❌")
-            all_checks_passed = False
-        else:
-            checks = pandas.DataFrame(
-                columns=['key', 'expected', 'found', 'pass']
-            )
-            for key, expected in self.expected_counts.items():
-
-                # Accept both concepts and attributes, but automatically
-                # translate lone concepts as meaning id or else unique_key if
-                # one of those was discovered in the data.
-                if key not in counted:
-                    if key in str_to_CONCEPT:
-                        concept = str_to_CONCEPT[key]
-                        if concept.ID in counted:
-                            key = concept.ID
-                        elif concept.UNIQUE_KEY in counted:
-                            key = concept.UNIQUE_KEY
-
-                found = uniques.get(key)
-                if expected == found:
-                    passed = '✅'
-                else:
-                    all_checks_passed = False
-                    passed = '❌'
-                checks = checks.append(
-                    {
-                        'key': key, 'expected': expected, 'found': found,
-                        'pass': passed
-                    },
-                    ignore_index=True
-                )
-            message.append(
-                tabulate(
-                    checks,
-                    headers='keys', showindex=False, tablefmt='psql'
-                )
-            )
-
-        return all_checks_passed, counted, '\n'.join(message)
+                    sources[key][val].add(data_file)
+            for keyA in df.columns:
+                for keyB in df.columns:
+                    if keyB != keyA:
+                        for i in range(len(df)):
+                            links[keyA + keyB][df[keyA].iloc[i]].add(
```

Comment:
Can you add a delimiter to the link key? A key looks like this right now and its tough to read: 
```
"CONCEPT|BIOSPECIMEN|ALIQUOT_IDCONCEPT|BIOSPECIMEN|ANALYTE"
```

Reply:
√